apps/api/src/app/blueprint/usecases/get-grouped-blueprints/get-grouped-blueprints.usecase.ts

```diff
@@ -14,14 +17,48 @@ export class GetGroupedBlueprints {
     builder: () => buildGroupedBlueprintsKey(),
     options: { ttl: WEEK_IN_SECONDS },
   })
-  async execute(): Promise<GroupedBlueprintResponse[]> {
-    const blueprints = await this.notificationTemplateRepository.findAllGroupedByCategory();
-    if (!blueprints) {
+  async execute(): Promise<GroupedBlueprintResponse> {
+    const groups = await this.fetchGroupedBlueprints();
+
+    const updatePopularBlueprints = this.updatePopularBlueprints(groups);
+
+    const popular = { name: POPULAR_TEMPLATES_GROUPED.name, blueprints: updatePopularBlueprints };
+
+    return { general: groups as GroupedBlueprint[], popular };
+  }
+
+  private async fetchGroupedBlueprints() {
+    const groups = await this.notificationTemplateRepository.findAllGroupedByCategory();
+    if (!groups) {
       throw new NotFoundException(
         `Blueprints for organization id ${NotificationTemplateRepository.getBlueprintOrganizationId()} were not found`
       );
     }
 
-    return blueprints;
+    return groups;
+  }
+
+  private extractStoredBlueprints(groupes: { name: string; blueprints: NotificationTemplateEntity[] }[]) {
+    return groupes
+      .map((group) => group.blueprints)
+      .reduce((acc, curr) => {
+        acc.push(...curr);
+
+        return acc;
+      });
```

Comment:
I don't get why you need to reduce an array to convert it into an array of its own elements. Don't know what nuance I am missing.

Reply:
You are not missing a thing, just thought it will be easier to work with array on the next step 😅

Reply:
Ah, I knew I was missing something. 😄 
What about using prototype's `flat` instead ?